src/main/java/com/google/devtools/build/lib/analysis/config/HostTransition.java

```diff
@@ -50,4 +53,37 @@ public BuildOptions patch(BuildOptions options) {
       return options.createHostOptions();
     }
   }
+
+  /** Returns a {@link TransitionFactory} instance that generates the host transition. */
+  public static <T extends TransitionFactoryData> TransitionFactory<T> createFactory() {
+    return new AutoValue_HostTransition_Factory<>();
```

Comment:
AutoValue will automatically make .equals() and .hashCode() for these, right?

If we start throwing these things into hash maps, etc. before the TransitionFactory -> Transition transformation, we'll have to be careful that the same requirements for those methods on Transition apply here.

Reply:
Yes, AutoValue creates meaningful hashCode, equals, and toString methods.